enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,213 @@
+  - "@abhinavdahiya"
+  - "@bparees"
+  - "@dhansen"
+  - "@sdodson"
+approvers:
+  - TBD
+creation-date: 2019-11-14
+last-updated: 2019-11-14
+status: provisional
+see-also:
+  - "/enhancements/automated-service-ca-rotation.md"
+  - "/enhancements/kube-apiserver/certificates.md"
+  - "/enhancements/kube-apiserver/tls-config.md"
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"
+---
+
+# X.509 Trust
+
+This enhancement provides a big-picture overview of X.509 trust management for cluster components.
+This enhancement provides for both cluster-scoped defaults and per-component overrides.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Several cluster components may make TLS requests to external services.
+For example, [the cluster-version operator][cluster-version-operator] may request available updates, [the machine-API operator][machine-api-operator] may call platform APIs to manage machines, [CRI-O][cri-o] may contact container image registries when downloading images for pods, etc.
+Cluster components may also perform other X.509 verification besides TLS requests.
+For example, user mail processing components may perform [S/MIME][smime] verification.
+This enhancement provides a framework for managing X.509 trust for these situations.
+
+Currently the default trust for CRI-O, the kubelet's cloud provider, and other node-namespace egress is the node's X.509 certificate authority (CA) store, which depends on the operating system ([Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], etc.) and any subsequent configuration ([the machine-config operator][machine-config-operator], etc.).
+The default trust for containers like the machine-API operator is that container's CA store, which depends on the base layer ([Red Hat Universal Base Image (UBI)][ubi], etc.).
+Many external components will have X.509 certificates signed by one of those default CAs, but this is not always the case.
+
+There may also be internal communication between cluster components that relies on X.509 trust but lacks a custom 
+The framework from this enhancement can also be used to configure trust for that communication as well.
+
+This enhancement provides a mechanism for configuring additional or alternative trust bundles for all cluster services that do not already have existing trust configuration mechanisms.
+
+## Motivation
+
+### Goals
+
+Administrators will have a single location to configure a default additional trust bundle to be used by all cluster components.
+They will also be able to specify whether this default is to be use in addition to, or instead of, the component's default trust bundle.
+
+Administrators will have per-component configuration locations to adjust the cluster-scoped default.
```

Comment:
Why?

Reply:
> Why?

Because you don't need to have a global trust store for the whole cluster.  You get better isolation against compromised CAs if each component only trusts the CAs it needs to trust (e.g. the CVO scoping I sketched out [here][1]).  But while that scoping is nice for the paranoid, it's also a pain, which is why I think we also want to make it easy for folks to mix and match cluster-scoped defaults and per-component overrides to match their security stance.

[1]: https://github.com/openshift/enhancements/pull/115#discussion_r349951652